packages/datadog_flutter_plugin/lib/src/rum/ddrum_events.dart

```diff
@@ -0,0 +1,697 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022-Present Datadog, Inc.
+
+import 'package:json_annotation/json_annotation.dart';
+
+import '../../datadog_flutter_plugin.dart';
+import '../json_helpers.dart';
+
+part 'ddrum_events.g.dart';
+
+// Currently excluded
+//  * ciTest
+//  * display
+//  * featureFlags
+//  * source
+//  * synthetics
+@commonJsonOptions
+class RumViewEvent {
```

Comment:
side question: what if we write a generator for these classes from the schema, the same way we do for iOS/Android. Do you think it is worth it? Probably it will be a little of added value, because most of the properties added are read-only, but I'm wondering about your opinion.

Reply:
I thought about it, but I'm concerned about the level of effort versus the payoff. If I have to add a lot more events in the future I'll look into the generator.